tests/checks/message-localization.fish

```diff
@@ -104,3 +117,71 @@ echo (_ file)
 set -l LC_ALL de_DE.utf8
 echo (_ file)
 # CHECK: Datei
+
+# Check `status language` builtin
+set --erase LANG
+set --erase LC_MESSAGES
+set --erase LC_ALL
+set --erase LANGUAGE
+status language
+# CHECK: Active languages (default):
+echo (_ file)
+# CHECK: file
+
+set -l LANGUAGE pt_BR de_DE
+status language
+# CHECK: Active languages (from variable LANGUAGE): pt_BR de
+echo (_ file)
+# CHECK: arquivo
+
+# We have fr but not fr_FR. For the builtin command, only exact matches are allowed.
+status language set fr_FR de pt_BR
+# CHECKERR: No catalogs available for language specifiers: fr_FR
+status language
+# CHECK: Active languages (from command `status language set`): de pt_BR
+echo (_ file)
+# CHECK: Datei
+
+set -l LANGUAGE zh_TW
+status language
+# CHECK: Active languages (from command `status language set`): de pt_BR
+echo (_ file)
+# CHECK: Datei
+
+set -l LC_MESSAGES C
+status language
+# CHECK: Active languages (from command `status language set`): de pt_BR
+echo (_ file)
+# CHECK: Datei
+
+status language unset
+status language
+# CHECK: Active languages (from variable LC_MESSAGES):
```

Comment:
the "from" phrasing is maybe not perfect, maybe something like "source: " would work better.

I'll queue this now since this is not worse than my suggestion, and I guess we don't have further changes planned.